packages/codemirror/src/editor.ts

```diff
@@ -615,7 +631,17 @@ class CodeMirrorEditor implements CodeEditor.IEditor {
       return;
     }
     this._changeGuard = true;
-    this._model.value.text = this.doc.getValue();
+    let value = this._model.value;
+    let start = doc.indexFromPos(change.from);
```

Comment:
Okay, I have applied the change in the pre-change coordinate system, which in my testing seems to fix #1828 .